### PR TITLE
[JENKINS-69306] fix: add the System Info menu item to allow administrator to view agent properties when running

### PR DESCRIPTION
**Pull request by dduportal**
With the help of @jglick , I was able to propose this patch.

Opening as draft to check the CI result, while I'm trying to understand how to package, run and manually test this locally.

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly [hunk 1]
```diff
@@ -27,4 +27,5 @@ THE SOFTWARE.
   <l:task icon="icon-clipboard icon-md" href="${rootURL}/${it.url}log" title="${%Log}" permission="${app.ADMINISTER}"/>
```
Comment (jglick) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly:27]:
```suggestion
  <l:task icon="icon-clipboard icon-md" href="${rootURL}/${it.url}log" title="${%Log}" permission="${it.EXTENDED_READ_AND_CONNECT}"/>
```
https://github.com/jenkinsci/jenkins/blob/3ba60ab3135773e8e8e58cc0dc2fdbda10a32428/core/src/main/resources/hudson/slaves/SlaveComputer/sidepanel2.jelly#L27

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly [hunk 2]
```diff
@@ -27,4 +27,5 @@ THE SOFTWARE.
...
   <l:task icon="icon-computer icon-md" href="${rootURL}/${it.url}podLog" title="${%Pod Log}" permission="${app.ADMINISTER}"/>
   <l:task icon="icon-computer icon-md" href="${rootURL}/${it.url}events" title="${%Events}" permission="${app.ADMINISTER}"/>
+  <l:task icon="icon-computer icon-md" href="${rootURL}/${it.url}systemInfo" title="${%System Information}" permissions="${it.EXTENDED_READ_AND_CONNECT}"/>
```
Comment (jglick) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly:30]:
```suggestion
  <j:if test="${it.channel!=null}">
    <l:task icon="icon-computer icon-md" href="${rootURL}/${it.url}systemInfo" title="${%System Information}" permissions="${it.EXTENDED_READ_AND_CONNECT}"/>
  </j:if>
```
https://github.com/jenkinsci/jenkins/blob/3ba60ab3135773e8e8e58cc0dc2fdbda10a32428/core/src/main/resources/hudson/slaves/SlaveComputer/sidepanel2.jelly#L28

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly [hunk 3]
```diff
@@ -27,4 +27,5 @@ THE SOFTWARE.
   <l:task icon="icon-clipboard icon-md" href="${rootURL}/${it.url}log" title="${%Log}" permission="${app.ADMINISTER}"/>
   <l:task icon="icon-computer icon-md" href="${rootURL}/${it.url}podLog" title="${%Pod Log}" permission="${app.ADMINISTER}"/>
   <l:task icon="icon-computer icon-md" href="${rootURL}/${it.url}events" title="${%Events}" permission="${app.ADMINISTER}"/>
```
Comment (jglick) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly:29]: probably

```suggestion
  <l:task icon="icon-computer icon-md" href="${rootURL}/${it.url}podLog" title="${%Pod Log}" permission="${it.EXTENDED_READ_AND_CONNECT}"/>
  <l:task icon="icon-computer icon-md" href="${rootURL}/${it.url}events" title="${%Events}" permission="${it.EXTENDED_READ_AND_CONNECT}"/>
```
as well, though then https://github.com/jenkinsci/kubernetes-plugin/blob/c3f2a06be34e69614cc8de72340f63cc78f8fbe7/src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/podLog.jelly#L27 and https://github.com/jenkinsci/kubernetes-plugin/blob/c3f2a06be34e69614cc8de72340f63cc78f8fbe7/src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/events.jelly#L27 would need to be changed to match

**Review by jglick — COMMENTED** [review]
BTW I wonder if https://github.com/jenkinsci/kubernetes-plugin/blob/c3f2a06be34e69614cc8de72340f63cc78f8fbe7/src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/container.jelly#L34 is actually displayed somewhere?

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly (hunk 3 above)
Comment (dduportal) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly]: Done in https://github.com/jenkinsci/kubernetes-plugin/pull/1228/commits/5c620515ce125787015532a2ce0f0640f99a641a

**Comment by dduportal** [comment]
> BTW I wonder if
> 
> https://github.com/jenkinsci/kubernetes-plugin/blob/c3f2a06be34e69614cc8de72340f63cc78f8fbe7/src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/container.jelly#L34
> is actually displayed somewhere?

From the pod agent's page, if you click on the "Pod Log" side menu, it shows a table of the pod's containers (with no logs).
If you click on one of the containers (for instance `jnlp`) , then you are sent to a page with the container logs which has a spinner. See attached screenshots.

I might be wrong (could another jelly template from another plugin) but since this is the only occurence of `spinner.gif` in this code base, I assume it is this page (gotta check on my local tests).

Capture d’écran 2022-08-11 à 11 46 24
Capture d’écran 2022-08-11 à 11 46 32

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly (hunk 3 above)
Comment (jglick) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly]: Only the Jelly parts, but you would also need to adjust the Java permissions to match, or under some conditions you would be exposing links which a user could click on only to get a 403.

**Comment by jglick** [comment]
> you are sent to a page with the container logs

Oh, that is handy for debugging. Better IMO would be for this content to be copied to the build log to begin with, with timestamps interleaved with other messages and accessible from any build log viewer and not just the Jenkins GUI. Anyway off topic here, was just curious whether that Jelly page was still in use or dead code.

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly (hunk 3 above)
Comment (dduportal) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly]: Oh, I misunderstood the suggestion. I need help on this one then as I have no idea what to do and where to look.

**Comment by dduportal** [comment]
> > you are sent to a page with the container logs
> 
> Oh, that is handy for debugging. Better IMO would be for this content to be copied to the build log to begin with, with timestamps interleaved with other messages and accessible from any build log viewer and not just the Jenkins GUI. Anyway off topic here, was just curious whether that Jelly page was still in use or dead code.

Ack for the off topic, but that was a good opportunity to mention it, thanks!

By the way, your suggestion looks awesome. But I'm not sure if it can be done easily since the container logs are streamed; is it possible to "mix" the build log streams with the underlying container agent logs?

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly (hunk 3 above)
Comment (jglick) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/sidepanel2.jelly]: Sorry, I got confused, this is not about checks from Java code and you did do this.

**Comment by jglick** [comment]
> is it possible to "mix" the build log streams with the underlying container agent logs?

We already do so for the K8s event logs, or did until #1192. Not a problem from the Jenkins build log API perspective, just a question of actually getting this information from the API server efficiently.

**Review by jglick — APPROVED** [review]

**Comment by dduportal** [comment]
In the curent state, the "System Information" panel did replace the pod specific (logs, event, etc.):

Capture d’écran 2022-08-12 à 14 40 48

I'm playing around to see how it behaves

**Comment by dduportal** [comment]
If I change `permission="${app.ADMINISTER}"` to `permission="${app.ADMINISTER}"` then it behaves as expected 🤔 
I'm not sure were would be the reference documentation for Jelly templates in the context of Jenkins?

**Comment by dduportal** [comment]
With https://github.com/jenkinsci/kubernetes-plugin/pull/1228/commits/3d3cd367c5c242277f13c91b9fab99fca82d7fd0, I got the following behavior:

Capture d’écran 2022-08-12 à 15 10 57

**Comment by jglick** [comment]
> If I change `permission="${app.ADMINISTER}"` to `permission="${app.ADMINISTER}"` then it behaves as expected

I think you mean from `it.EXTENDED_READ_AND_CONNECT`. Then something is wrong. You should use the permission scheme used in core. Is this newer than the current `jenkins.version` (2.332.1)?

> the reference documentation for Jelly templates in the context of Jenkins

Nothing that is going to help you.

You can interactively debug by writing stuff in Jelly like

```
app=${app} app.ADMINISTER=${app.ADMINISTER}
it=${it} it.EXTENDED_READ_AND_CONNECT=${it.EXTENDED_READ_AND_CONNECT}
hP(ADM)=${it.hasPermission(app.ADMINISTER)}
hP(ERC)=${it.hasPermission(it.EXTENDED_READ_AND_CONNECT)}
```

**Comment by dduportal** [comment]
> I think you mean from it.EXTENDED_READ_AND_CONNECT.

Yes. My bad I messed up the copy and paste. I meant: `If I change permission="${app.ADMINISTER}" to permission="${it.EXTENDED_READ_AND_CONNECT}"` which is the commit https://github.com/jenkinsci/kubernetes-plugin/commit/3d3cd367c5c242277f13c91b9fab99fca82d7fd0.

> You should use the permission scheme used in core. Is this newer than the current jenkins.version (2.332.1)?

I'm using the following setup: https://github.com/dduportal/jenkins-examples/blob/8f0bc983389038d90a2b61bed571105069e5c162/kubernetes/helmfile.yaml#L20-L25 
which uses the latest LTS as per https://github.com/dduportal/jenkins-examples/blob/main/docker-image/Dockerfile#L1.
=> It was the version `2.346.3` (same as ci.jenkins.io).

**Comment by jglick** [comment]
By _this_ I mean the idioms currently used in `SlaveComputer/entries2.jelly`.`EXTENDED_READ_AND_CONNECT` was introduced in the past year or two IIRC.

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/container.jelly [hunk 4]
```diff
@@ -24,7 +24,7 @@ THE SOFTWARE.
 
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt">
-  <l:layout title="${it.displayName} log" permission="${app.ADMINISTER}">
+  <l:layout title="${it.displayName} log" permission="${it.EXTENDED_READ_AND_CONNECT}">
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/container.jelly:27]: Change is inconsistent with the sidebar links, which still require ADMINISTER.

Comment (dduportal) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/container.jelly:27]:
```suggestion
  <l:layout title="${it.displayName} log" permission="${app.ADMINISTER}">
```

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/events.jelly [hunk 5]
```diff
@@ -24,7 +24,7 @@ THE SOFTWARE.
 
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt">
-  <l:layout title="${it.displayName} log" permission="${app.ADMINISTER}">
+  <l:layout title="${it.displayName} log" permission="${it.EXTENDED_READ_AND_CONNECT}">
```
Comment (dduportal) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/events.jelly:27]:
```suggestion
  <l:layout title="${it.displayName} log" permission="${app.ADMINISTER}">
```

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/podLog.jelly [hunk 6]
```diff
@@ -24,7 +24,7 @@ THE SOFTWARE.
 
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt">
-  <l:layout title="${it.displayName} log" permission="${app.ADMINISTER}">
+  <l:layout title="${it.displayName} log" permission="${it.EXTENDED_READ_AND_CONNECT}">
```
Comment (dduportal) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/podLog.jelly:27]:
```suggestion
  <l:layout title="${it.displayName} log" permission="${app.ADMINISTER}">
```

**Review by Vlatombe — APPROVED** [review]

**Pull request merged**